framework/src/meshgenerators/MeshExtruderGenerator.C

```diff
@@ -15,7 +15,7 @@
 #include "libmesh/boundary_info.h"
 #include "libmesh/elem.h"
 
-registerMooseObject("MooseApp", MeshExtruderGenerator);
+registerMooseObjectDeprecated("MooseApp", MeshExtruderGenerator, "12/31/2025 24:00");
```

Comment:
why is this being deprecated?

Reply:
people try to use it and then look for the features that are only in the advanced extruder generator.

in general two objects with the same functionality is not a good idea
I would support it if we added the Simple... prefix to all these objects that are desired to just do one simple thing. we're not doing that though